tmail-backend/mailbox/team-mailboxes/src/test/scala/com/linagora/tmail/team/TeamMailboxRepositoryContract.scala

```diff
@@ -0,0 +1,198 @@
+package com.linagora.tmail.team
+
+import com.google.common.collect.ImmutableSet
+import com.linagora.tmail.team.Namespace.TEAM_MAILBOX_NAMESPACE
+import com.linagora.tmail.team.TeamMailboxRepositoryContract.{ANDRE, BOB, TEAM_MAILBOX, TEAM_MAILBOX_2}
+import org.apache.james.core.Username
+import org.apache.james.mailbox.inmemory.{InMemoryMailboxManager, MemoryMailboxManagerProvider}
+import org.apache.james.mailbox.model.MailboxACL
+import org.apache.james.mailbox.{MailboxManager, MailboxSession, SessionProvider}
+import org.assertj.core.api.Assertions.{assertThat, assertThatCode, assertThatThrownBy}
+import org.junit.jupiter.api.{BeforeEach, Test}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.util
+import scala.jdk.CollectionConverters._
+
+object TeamMailboxRepositoryContract {
+  val TEAM_MAILBOX_USER: Username = Username.of("teammail1@linagora.com")
+  val TEAM_MAILBOX_USER_2: Username = Username.of("teammail2@linagora.com")
+  val TEAM_MAILBOX: TeamMailbox = TeamMailbox(TEAM_MAILBOX_NAMESPACE, TEAM_MAILBOX_USER, MailboxName("Sent"))
+  val TEAM_MAILBOX_2: TeamMailbox = TeamMailbox(TEAM_MAILBOX_NAMESPACE, TEAM_MAILBOX_USER_2, MailboxName("Sent"))
+  val BOB: Username = Username.of("bob@linagora.com")
+  val ANDRE: Username = Username.of("andre@linagora.com")
+}
+
+trait TeamMailboxRepositoryContract {
+
+  def testee: TeamMailboxRepository
+
+  def mailboxManager: MailboxManager
+
+  def sessionProvider: SessionProvider
+
+  @Test
+  def createTeamMailboxShouldStoreAssignMailbox(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    assertThat(SMono.fromPublisher(mailboxManager.mailboxExists(TEAM_MAILBOX.mailboxPath,
+      sessionProvider.createSystemSession(TEAM_MAILBOX.tmUser)))
+      .block())
+      .isTrue
+  }
+
+  @Test
+  def createTeamMailboxShouldThrowWhenAssignMailboxExists(): Unit = {
```

Comment:
IMO it would be nicer to be idempotent (what if we create 1 mailbox out of the 3?)

Reply:
We can catch mailboxExistsException. `onErrorResume` blah blah...